sources/packages/api/src/route-controllers/student/student.controller.ts

```diff
@@ -318,25 +318,30 @@ export class StudentController extends BaseController {
     }));
   }
   /**
+   * TODO: This api will be called by UI and update states, in future restriction UI ticket
    * GET API which returns student restriction details.
-   * @param userToken
-   * @returns Student Restriction
+   * @param studentToken student token.
+   * @returns Student restriction code and notification type as key value pair, if any.
    */
   @AllowAuthorizedParty(AuthorizedParties.student)
   @Get("restriction")
   async getStudentRestrictions(
-    @UserToken() userToken: IUserToken,
-  ): Promise<StudentRestrictionDTO> {
-    const studentRestrictionStatus =
-      await this.studentRestrictionService.getStudentRestrictionsByUserId(
-        userToken.userId,
+    @UserToken() studentToken: StudentUserToken,
+  ): Promise<StudentRestrictionAPIOutDTO[]> {
```

Comment:
Maybe for the sake of finishing this PR, this method can be here but this entire file will be gone when the student reorganization API ticket is completed 😉

Reply:
@dheepak-aot, once I merge Ann's PR with mine, this will be moved to another controller. Probably it will be a minor change on your current ticket if I merge it first, just a heads up.